### PR TITLE
Show popups in editor when loading/saving file fails, fix minor issues with preview image

### LINKED ISSUE #6430: Add some indication why a file could not be loaded
Currently if a file fails to load the editor just acts like you never pressed the button. This was especially annoying when trying to convert a sound file to a compatible opus file.

### PR DESCRIPTION
**Pull request by Robyt3**
screenshot_2023-03-17_13-00-59

screenshot_2023-03-17_13-00-46

screenshot_2023-03-17_13-00-38

screenshot_2023-03-17_13-00-16

screenshot_2023-03-19_11-12-28

screenshot_2023-03-17_12-59-42

When using Ctrl+S:

screenshot_2023-03-17_12-59-31

Closes #6430.

## Checklist

- [X] Tested the change ingame
- [X] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [X] Considered possible null pointers and out of bounds array indexing
- [X] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/editor.cpp [hunk 1]
```diff
@@ -807,26 +807,43 @@ int CEditor::FindSelectedQuadIndex(int Index) const
...
 {
 	CEditor *pEditor = (CEditor *)pUser;
 	if(pEditor->Load(pFileName, StorageType))
 	{
 		pEditor->m_ValidSaveFilename = StorageType == IStorage::TYPE_SAVE && pEditor->m_pFileDialogPath == pEditor->m_aFileDialogCurrentFolder;
 		pEditor->m_Dialog = DIALOG_NONE;
+		return true;
+	}
+	else
+	{
```
Comment (def-) [inline src/game/editor/editor.cpp:820]: There seems to be a lot of duplication of this. Can we have `ShowFileDialogError` take a format string and it's arguments as varargs and then do the `str_format` internally?

**Review by def- — APPROVED** [review]

**Review by def- — APPROVED** [review]
bors r+
Thanks

**Comment by Robyt3** [comment]
Added the missing include.

**Pull request merged**